core/src/main/java/org/infinispan/commands/functional/AbstractWriteManyCommand.java

```diff
@@ -30,13 +35,15 @@ protected AbstractWriteManyCommand(CommandInvocationId commandInvocationId,
       this.flags = params.toFlagsBitSet();
       this.keyDataConversion = keyDataConversion;
       this.valueDataConversion = valueDataConversion;
+      this.internalMetadataMap = new ConcurrentHashMap<>();
```

Comment:
no, otherwise I will get an `UnsupportedException` when I try to put metadata there.
It can be `null` but then I will have to put some `null` checks before get/put and do a proper synchronization around it. I would rather not do it.